releasenotes/notes/components-io-discoverability-0ecb53d47997a273.yaml

```diff
@@ -0,0 +1,67 @@
+---
+highlights: >
+  To ease discoverability of inputs and outputs we added two new fields to all Components:
+  - `inputs`
+  - `outputs`
+  This means those fields are now reserved and cannot be used in any `Component`.
+
+  An example usage would look like this:
+  ```python
+  from haystack.components.builders.prompt_builder import PromptBuilder
+
+  prompt_template = \"""
+  Given these documents, answer the question.\nDocuments:
+  {% for doc in documents %}
+      {{ doc.content }}
+  {% endfor %}
+
+  \nQuestion: {{question}}
+  \nAnswer:
+  \"""
+
+  prompt_builder = PromptBuilder(template=prompt_template)
+
+  prompt_builder.inputs
+  >>> PromptBuilder inputs:
+  >>>   - question: Any
+  >>>   - documents: Any
+
+  prompt_builder.inputs.question
+  >>> PromptBuilder.inputs.question
+  ```
+
+  The behaviour changes a bit when the `Component` has been added to a `Pipeline`. In that case we'll get
+  the name of that `Component` instance as it has been added to that `Pipeline`.
+
+  That way we can also use this fields when calling `Pipeline.connect()`.
```

Comment:
This is not true as of now since I need to update `Pipeline.connect()` to support the new syntax with `.inputs.` and `.outputs.`.

I'll keep it as it's a nice highlight to have, I won't add a release note on the PR that updates `Pipeline.connect()`.

Reply:
We already have autocompletion with this PR.